app/jobs/oauth2_refresh_job.rb

```diff
@@ -0,0 +1,20 @@
+# typed: true
+
+class Oauth2RefreshJob < ApplicationJob
+  queue_as :default
+
+  def perform(connection_id, klass_name)
+    case klass_name
+    when "UpholdConnection"
```

Comment:
This is a thought experiment and won't hold up the PR b/c it looks pretty good as is...

A thought experiment for (potential) simplification. 

If we raise this `case` up one level and call connection-specific refreshers, for ex `UpholdWalletRefreshJob`, `GeminiWalletRefreshJob`, etc.

Those classes could handle all of the details internally for updating a connection's refresh information. That would be the only contract. I call you, you refresh the token for your particular provider. You're free to do that however you want.

Those jobs could make the HTTP call in one line(`ClientCredentials`, then do the equivalent of what 
`Success.new(connection: connection.update_access_tokens!(result))` is doing now -- in one class. I _think_ it would be easier to read and see "oh, this is doing the 2-step process in here` -- just a thought.

But all in one class, there to see. Just an idea. We wouldn't need `oath2_refresh_job`, or `client_credentials` becomes that...

This gives us a little flexibility for the provider-specific wallet providers to customize as needed, but they are still able to compose the lower level pieces. But we wouldn't force those lower level pieces to conform to the http specifics of each provider.

Reply:
I don't want to have 3 separate classes that implement the same refresh token flow when I can write an HTTP client, a service, and an interface that implements an API that can be used for all three cases/potentially all cases.

Let me show you the power of OOP.  COME TO THE DARK SIDE TYLER.

Reply:
I considered converting OAuth2RefreshJob into <wallet type>RefreshJob, but they each end up being pretty much identical with only the names different.  I might end up doing that in the end because passing the klass_name is a bit awkward, but I didn't want to linger on it too much right now.  I'll likely end up modifying it as I continue with the the refresh token elements.  Everything I do is iterative and so I will refine different pieces as I go.